### PR TITLE
The rest of the Terra redirects [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
This PR builds off #1521. That other PR created the infrastructure to handle Terra redirects and added one redirect (workspace summary page) as a proof of concept. This PR adds the remainder of the redirects, with a few outstanding TODOs that are documented inline.

In discussion with QA, we will not be adding automated tests for these - the cost/benefit ratio is not worth it.

This document lists all of the redirects, with a few highlighted as needing attention/waiting on dependencies: https://docs.google.com/spreadsheets/d/1qnXj71_FhYwsPS7kbq1grWH8b9nhUoQ_7IAOMrWdzVs/edit?usp=sharing

UPDATE: most recent commit adds a redirect for billing project detail, and moves the querystring after the fragment in the redirect urls. Thus, we will have
`/#workspaces?fcredir=1` instead of `/?fcredir=1#workspaces`. This is incorrect according to standards bodies but is the format that Terra UI expects; Terra will change this expectation in a future PR.

-----

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/billing/billing_management.cljs [hunk 1]
```diff
@@ -189,11 +189,15 @@
...
    {:component Page
     :regex #"billing"
+    :terra-redirect #(str "billing")
     :make-props (fn [_] {})
     :make-path (fn [] "billing")})
   (nav/defpath
    :billing-project
    {:component Page
     :regex #"billing/([^/]+)"
+     ;; Terra UI cannot direct-link to an individual billing project. So, if a FireCloud user has a bookmark/link
```
Comment (elizzhao) [inline src/cljs/main/broadfcui/page/billing/billing_management.cljs]: I think we can link to individual billing projects in Terra.

**Review by elizzhao — COMMENTED** [review]
Checked the spreadsheet and the billing project PR was merged, so you should be able to directly link to it. Otherwise, I think it looks good

File: src/cljs/main/broadfcui/page/billing/billing_management.cljs [hunk 2]
```diff
@@ -189,11 +189,13 @@
...
    {:component Page
     :regex #"billing"
+    :terra-redirect #(str "billing")
     :make-props (fn [_] {})
     :make-path (fn [] "billing")})
   (nav/defpath
    :billing-project
    {:component Page
     :regex #"billing/([^/]+)"
+    :terra-redirect (fn [args] (str "billing?type=project&selectedName=" (:project-name args)))
```
Comment (elizzhao) [inline src/cljs/main/broadfcui/page/billing/billing_management.cljs]: So it doesn't appear to cause a problem (in that the url works either way) but note that the order is switched between `type=project` and `selectedName=` in this url compared to the one generated in Terra

**Comment by davidangb** [comment]
auto tests had one failure in `IGV should render on the page`, which is unrelated. Nonetheless: jenkins retest

**Review by elizzhao — APPROVED** [review]
Looks good!

**Pull request merged**